### ISSUE TITLE
Problems with DITA 1.3 XML Schema-based specializations

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2124]
1.  DITA 1.2 XML Based Subject Scheme maps refer to the XML Schema like:
   
   ``` xml
   <subjectScheme xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:oasis:names:tc:dita:spec:classification:xsd:subjectScheme.xsd">
   ```
   
   This `urn:oasis:names:tc:dita:spec:classification:xsd:subjectScheme.xsd` URN no longer seems to be mapped anywhere in the calalogs. It seems to have been replaced with the URN mapping `urn:oasis:names:tc:dita:xsd:subjectScheme.xsd`. But this change no longer insures backward compatibility.
2.  Even if I create a Subject Scheme map with the new URN `urn:oasis:names:tc:dita:xsd:subjectScheme.xsd`, validation complains that the @domains attribute is not defined on the root element.
3.  A DITA 1.2 XML Based glossgroup referenced the following schema URN:
   
   ``` xml
   <glossgroup id="glossgroup_kzj_fyn_wt" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:oasis:names:tc:dita:xsd:glossarygroup.xsd">
   ```
   
   This URN no longer seems to be available in the XML catalog mapping. Instead right now there is a catalog mapping for `urn:oasis:names:tc:dita:xsd:glossgroup.xsd`. Ideally the original mapping would also work for backward compatibility reasons.

**Comment by jelovirt** [comment]
Since DITA-OT uses original DITA 1.3 grammar files and catalogs, this is a DITA grammar file issue and should be raised with the TC.

cc @drmacro @robander @keberlein

**Comment by raducoravu** [comment]
Right, I figured that Eliot will also need an issue on which to open a pull request.

**Comment by raducoravu** [comment]
By the way, if the DITA 1.3 specs officially supports only the RNG schemas (the others being offered on the side), then this would not actually be something that the TC would need to fix, it would be just a conversion problem.

**Comment by keberlein** [comment]
Radu, please send an e-mail to the dita-comment list. Thank you.

**Comment by drmacro** [comment]
The schemas used (and must continue to use) inconsistent patterns for URIs and at least at one point I was using the incorrect URIs for the schemas but I thought I had corrected it.

I will check the URIs and catalogs and make sure that they match the original 1.2 values.

**Comment by drmacro** [comment]
I have updated the subject scheme RNG and XSD files to add "spec:classification:" to all URIs. I compared against the original 1.2 XSDs. 

I had corrected (or done the right thing) for classifyMap and classifyDomain but apparently missed doing it for subject scheme also.

**Comment by raducoravu** [comment]
Great Eliot, where can I get the fixes from? Or will there be a pull request?

**Comment by drmacro** [comment]
The fixes are committed to the TC SVN repo. I presume Robert will prepare an official grammar package update at some point.

**Comment by keberlein** [comment]
Robert and I will prepare a new package for the new stage in the
    OASIS process later this week. Is that timely enough for you, Radu?
    
    Best,
      Kris
      
      Kristen James Eberlein
      Chair, OASIS DITA Technical Committee
      Principal consultant, Eberlein Consulting
      www.eberleinconsulting.com
      +1 919 682-2290; kriseberlein (skype)
      
    
    On 11/3/2015 8:01 AM, Eliot Kimber
      wrote:
    
    
      The fixes are committed to the TC SVN repo. I presume Robert
        will prepare an official grammar package update at some point.
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by raducoravu** [comment]
It's OK, I'm in no hurry + I can also take a peak inside the TC SVN repository.

**Comment by jelovirt** [comment]
@robander @keberlein will you be able to build the grammar packages, or send a PR fixing the issues?

**Comment by keberlein** [comment]
Hi, Jarno.
    
    We are waiting on Eliot Kimber to complete the fix for X.
    
    As soon as that fix is in the DITA TC SVN repository, we'll build a
    new package.
    
    Best,
      Kris
      
      Kristen James Eberlein
      Chair, OASIS DITA Technical Committee
      Principal consultant, Eberlein Consulting
      www.eberleinconsulting.com
      +1 919 682-2290; kriseberlein (skype)
      
    
    On 11/6/2015 1:32 PM, Jarno Elovirta
      wrote:
    
    
      @robander @keberlein
        will you be able to build the grammar packages, or send a PR
        fixing the issues?
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by raducoravu** [comment]
Were there any commits on this issue?

**Comment by jelovirt** [comment]
87ac1a421a88e1eecf2d563d71504723f1d5f629 and c60608ea9e8edb4b34ff92b2be5ac47c5b25f52e contain the fix.

**Comment by raducoravu** [comment]
Could this be reopened?

Tested with DITA OT 2.2.1 and the glossgroup URI "urn:oasis:names:tc:dita:xsd:glossarygroup.xsd" is resolved in the catalog file "plugins/org.oasis-open.dita.v1_3\schema\technicalContent\catalog.xml" to a non-existing XSD locations (they should be prefixed by "xsd/").
Also for DITA 1.3 XML Schema-based Subject Scheme maps, validation still complains about a missing @domains attribute on the root element. This seems to happen because the XSD: "plugins\org.oasis-open.dita.v1_3\schema\subjectScheme\xsd\subjectSchemeMod.xsd" in the attribute group "subjectScheme.attributes" does not have a reference to the "domains-att".

**Comment by drmacro** [comment]
I'm looking at these now. Will report the bugs to the TC and correct in the TC SVN repo.

**Comment by raducoravu** [comment]
Great, thanks.

**Comment by drmacro** [comment]
Yesterday I also found a missing catalog entry in the base/dtd/catalog.xml for topicDefn.ent. This affects the D4P DTDs which directly refer to this file (which is probably not good practice but it's how I did it many years ago). I've reported that bug as well and have fixed it locally for commit to OASIS SVN pending TC approval.

**Comment by robander** [comment]
@drmacro just noticed this when searching for another issue. Fixed in 2.2 but reopened with items you and Radu noticed in January. Do you know if the fixes you expected for this item have gone to OASIS, and are part of the planned errata? Anything we should be fixing in 2.3.2?

**Comment by PStellmann** [comment]
In schema/base/catalog.xml I had to add some entries that were present as <system> but not as <uri>:
```
<uri name="urn:oasis:names:tc:dita:xsd:xml.xsd:1.3" uri="xsd/xml.xsd"/>
<uri name="urn:oasis:names:tc:dita:xsd:ditaarch.xsd:1.3" uri="xsd/ditaarch.xsd"/>
```

Otherwise I get an error when validating against embedded schematron with Saxon.

I also think that the existing system entries at the bottom should go up to the group with the other system entries.

**Comment by drmacro** [comment]
This error has already been corrected as an errata by the DITA TC and will be reflected in the updated schemas once the errata approval process is complete.

**Comment by PStellmann** [comment]
Thanks for the hint.

Another issue:
With the current DITA 1.3 schema it seems to be impossible to define a constraint to the enumerations of @type for the note element that works with the current version of Xerces (and, thus, with DITA-OT).

Since xs:redefine is more difficult with anonymous types in general I think it would be reasonable to re-add the explicite type "type-note-atts.class" that was used in 1.2!?

BTW: Is there a better place to discuss issues regarding these OASIS issues and/or take a look at the latest version?

**Comment by keberlein** [comment]
If you are not a member of the DITA Technical Committee, you
      should use the dita-comment list to communicate with the DITA TC.
      Here is a link that explains how you can subscribe to the
      dita-comment list:
    https://www.oasis-open.org/committees/comments/index.php?wg_abbrev=dita
    
    Best,
      Kris
      
      Kristen James Eberlein
      Chair, OASIS DITA Technical Committee
      Principal consultant, Eberlein Consulting
      www.eberleinconsulting.com
      +1 919 682-2290; kriseberlein (skype)
      
    
    On 8/10/2016 3:02 AM, Patrik Stellmann
      wrote:
    
    
      Thanks for the hint.
      Another issue:
        With the current DITA 1.3 schema it seems to be impossible to
        define a constraint to the enumerations of @type for the note element that works
        with the current version of Xerces (and, thus, with DITA-OT).
      Since xs:redefine is more difficult with anonymous types in
        general I think it would be reasonable to re-add the explicite
        type "type-note-atts.class" that was used in 1.2!?
      BTW: Is there a better place to discuss issues regarding these
        OASIS issues and/or take a look at the latest version?
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/dita-ot/dita-ot","title":"dita-ot/dita-ot","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/dita-ot/dita-ot"}},"updates":{"snippets":[{"icon":"PERSON","message":"@PStellmann in #2124: Thanks for the hint.\r\n\r\nAnother issue:\r\nWith the current DITA 1.3 schema it seems to be impossible to define a constraint to the enumerations of @type for the note element that works with the current version of Xerces (and, thus, with DITA-OT).\r\n\r\nSince xs:redefine is more difficult with anonymous types in general I think it would be reasonable to re-add the explicite type \"type-note-atts.class\" that was used in 1.2!?\r\n\r\nBTW: Is there a better place to discuss issues regarding these OASIS issues and/or take a look at the latest version?"}],"action":{"name":"View Issue","url":"https://github.com/dita-ot/dita-ot/issues/2124#issuecomment-238782714"}}}

**Comment by robander** [comment]
Closing - the remaining errors I see in the tread above (since this was re-opened) were addressed in #2860 in release 3.0.2. It looks like there were also duplicate issues open, which is probably how this one got missed when closing out related issues.
